modules/460-log-shipper/hooks/internal/vector/source/kubernetes.go

```diff
@@ -122,7 +122,7 @@ func (k Kubernetes) BuildSources() []impl.LogSource {
 	return res
 }
 
-func (k Kubernetes) MarshalJSON() ([]byte, error) {
+func (k *Kubernetes) MarshalJSON() ([]byte, error) {
```

Comment:
I don't think you need a pointer for Marshaller interface

Reply:
It's an awful practice because if you call `json.Marshal` on a non-pointer structure (`Kubernetes{}`) - this method will not be called. And it's not obvious for debugging.

You can find some example here - https://medium.com/hackernoon/today-i-learned-beware-with-pointer-in-golang-7764f2060a93

Reply:
I planned to move this part out of MarshalJSON, but forgot about it 🤦‍♂️ Will make a fix.

Reply:
Fixed.